src/main/java/frc/lib/swerve/MAXSwerve.java

```diff
@@ -0,0 +1,346 @@
+    // null);
+    // addChild("Gyro", m_gyro);
+    // addChild("frontLeft", m_frontLeft);
+    // addChild("frontRight", m_frontRight);
+    // addChild("rearLeft", m_rearLeft);
+    // addChild("rearRight", m_rearRight);
+    // addChild("Field 2d", m_field);
+  }
+
+  @Override
+  public void periodic() {
+    m_frontLeft.periodic();
+    m_frontRight.periodic();
+    m_backLeft.periodic();
+    m_backRight.periodic();
+
+    m_chassisSpeed = m_kinematics.toChassisSpeeds(getModuleStates());
+
+    m_odometry.update(
+        Rotation2d.fromDegrees(m_gyro.getAngle()),
+        new SwerveModulePosition[] {
+          m_frontLeft.getPosition(),
+          m_frontRight.getPosition(),
+          m_backLeft.getPosition(),
+          m_backRight.getPosition()
+        });
+    m_field.setRobotPose(m_odometry.getPoseMeters());
+  }
+
+  /**
+   * Return the current velocity of the chassis as a ChassisSpeeds object.
+   *
+   * @return velocity of the robot.
+   */
+  public ChassisSpeeds getChassisSpeeds() {
+    return m_chassisSpeed;
+  }
+
+  /**
+   * Predict the motion between the current position and a future state.
+   *
+   * @param lookahead Time in seconds to predict ahead.
+   * @return twist2d represnting the change in pose over the lookahead time.
+   */
+  public Twist2d getPredictedMotion(double lookahead) {
+    ChassisSpeeds chassisSpeed = m_kinematics.toChassisSpeeds(getModuleStates());
+    return new Twist2d(
+        chassisSpeed.vxMetersPerSecond * lookahead,
+        chassisSpeed.vyMetersPerSecond * lookahead,
+        chassisSpeed.omegaRadiansPerSecond * lookahead);
+  }
+
+  /**
+   * Return the modules positions
+   *
+   * @return SwerveModulePositions
+   */
+  public SwerveModulePosition[] getModulePositions() {
+    return m_ModulePositions;
+  }
+
+  /**
+   * Returns the currently-estimated pose of the robot.
+   *
+   * @return The pose.
+   */
+  public Pose2d getPose() {
+    return m_odometry.getPoseMeters();
+  }
+
+  /**
+   * Resets the odometry to the specified pose.
+   *
+   * @param pose The pose to which to set the odometry.
+   */
+  public void resetOdometry(Pose2d pose) {
+    m_odometry.resetPosition(m_gyro.getRotation2d(), getModulePositions(), pose);
+  }
+
+  /**
+   * Method to drive the robot using joystick info.
+   *
+   * @param xSpeed Speed of the robot in the x direction (forward).
+   * @param ySpeed Speed of the robot in the y direction (sideways).
+   * @param rot Angular rate of the robot.
+   * @param fieldRelative Whether the provided x and y speeds are relative to the field.
+   */
+  public void drive(double xSpeed, double ySpeed, double rot, boolean fieldRelative) {
+    // If nothing is commanded, hold the same position - don't know if the rev modules need this
+    // yet?
+    if (xSpeed == 0.0 && ySpeed == 0.0 && rot == 0.0) {
+      holdAllModulesRotation();
+      return;
+    }
+    // Adjust input based on max speed
+    xSpeed *= m_maxSpeed;
```

Comment:
Do you want to multiple xSpeed by m_maxSpeed?  Or limit it to it with xSpeed = min(xSpeed, m_maxSpeed)?
Same for ySpeed.